src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointComplications.java

```diff
@@ -0,0 +1,503 @@
+            // the reference system with a shorter overlap (i.e. with less-negative distance between regions) has a higher repeat number
+            resolveComplicationForComplexTandup(contigSeq, firstContigRegion, secondContigRegion, r1e, distBetweenAlignRegionsOnRef, distBetweenAlignRegionsOnCtg);
+        } else if (distBetweenAlignRegionsOnRef==0 && distBetweenAlignRegionsOnCtg==0) {// SNP & indel
+            throw new GATKException("Detected badly parsed chimeric alignment for identifying SV breakpoints; no rearrangement found: " + chimericAlignment.toString());
+        }
+
+        if (insertedSequenceForwardStrandRep.isEmpty()){
+            if (dupSeqRepeatNumOnCtg != dupSeqRepeatNumOnRef && null==dupSeqRepeatUnitRefSpan)
+                throw new GATKException("An identified breakpoint pair seem to suggest insertion but the inserted sequence is empty: " + chimericAlignment.toString());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void resolveComplicationForSimpleTandupExpansion(SimpleInterval leftReferenceInterval, byte[] contigSeq, AlignmentRegion firstContigRegion, AlignmentRegion secondContigRegion, int r1e, int r2b, int distBetweenAlignRegionsOnCtg) {
+        insertedSequenceForwardStrandRep = distBetweenAlignRegionsOnCtg==0 ? "" : getInsertedSequence(firstContigRegion, secondContigRegion, contigSeq); // note this does not incorporate the duplicated reference sequence
+        hasDuplicationAnnotation = true;
+        dupSeqRepeatUnitRefSpan          = new SimpleInterval(leftReferenceInterval.getContig(), r2b, r1e);
+        dupSeqRepeatNumOnRef      = 1;
+        dupSeqRepeatNumOnCtg      = 2;
+        cigarStringsForDupSeqOnCtg = new ArrayList<>(Collections.EMPTY_LIST);
+        if (firstContigRegion.forwardStrand) {
+            cigarStringsForDupSeqOnCtg.add( TextCigarCodec.encode(extractCigarForTandup(firstContigRegion, r1e, r2b)) );
+            cigarStringsForDupSeqOnCtg.add( TextCigarCodec.encode(extractCigarForTandup(secondContigRegion, r1e, r2b)) );
+        } else {
+            cigarStringsForDupSeqOnCtg.add( TextCigarCodec.encode(CigarUtils.invertCigar(extractCigarForTandup(firstContigRegion, r1e, r2b))) );
+            cigarStringsForDupSeqOnCtg.add( TextCigarCodec.encode(CigarUtils.invertCigar(extractCigarForTandup(secondContigRegion, r1e, r2b))) );
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void resolveComplicationForSimpleTandupContraction(SimpleInterval leftReferenceInterval, byte[] contigSeq, AlignmentRegion firstContigRegion, AlignmentRegion secondContigRegion, int r1e, int c1e, int c2b) {
+        homologyForwardStrandRep = getHomology(firstContigRegion, secondContigRegion, contigSeq);
+        hasDuplicationAnnotation = true;
+        cigarStringsForDupSeqOnCtg = new ArrayList<>(Collections.EMPTY_LIST);
```

Comment:
the default is to be in sync with `hasDuplicationAnnotation`: null when that is false